File: Practice/AllDomains/Tutorials/30DaysOfCode/Day24MoreLinkedLists.cs
/*	author:@shivkrthakur  */
using System;
using System.Collections.Generic;
class Node
{
	public int data;
	public Node next;
    public Node(int d){
        data=d;
        next=null;
    }
}

class Solution {
	public static Node removeDuplicates(Node head)
	{
		//Write your code here
		if(head != null)
		{
		 Node start = head;
		 while(start.next != null)
		 {
			 if(start.data == start.next.data)
			 {
				 if(start.next.next == null)
					 start.next = null;
				 else
					 start.next = start.next.next;
				 continue;
			 }
			 
			 start = start.next;
		 }
		}
	  return head;
	}
  
	public static  Node insert(Node head,int data)
	{
		Node p=new Node(data);
		if(head==null) head=p;
		
		else if(head.next==null) head.next = p;
		else
		{
			Node start=head;
			while(start.next!=null) start=start.next;
			start.next=p;
		}
		return head;
	}
	
	public static void display(Node head)
	{
		Node start=head;
		while(start!=null)
		{
			Console.Write(start.data+" ");
			start=start.next;
		}
	}
	
	static void Main(String[] args) 
	{
		Node head=null;
		int T = Int32.Parse(Console.ReadLine());
		while(T-->0)
		{
			int data=Int32.Parse(Console.ReadLine());
			head=insert(head,data);
		}
		head = removeDuplicates(head);
		display(head);
	}
}
	

File: Practice/AllDomains/Tutorials/30DaysOfCode/Day24MoreLinkedLists.js
/*	author:@shivkrthakur  */
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();
});
function readLine() {
    return input_stdin_array[input_currentline++];
}
function Node(data){
    this.data=data;
    this.next=null;
}
function Solution(){
this.removeDuplicates=function(head){
  //Write your code here
    var start = head;
    while(start){
        if(start.next)
        {
            if(start.data === start.next.data){
                start.next = start.next.next;
                continue;
            }
        }
        start = start.next;
    }
    return head;
}
this.insert=function(head,data){
      var p=new Node(data);
      if(head==null){
          head=p;
      }
      else if(head.next==null){
          head.next=p;
      }
      else{
          var start=head;
          while(start.next!=null){
              start=start.next;
          }
          start.next=p;
      }
      return head;

  };

this.display=function(head){
      var start=head;
          while(start){
              process.stdout.write(start.data+" ");
              start=start.next;
          }
  };
}

function main(){
  var T=parseInt(readLine());
  var head=null;
  var mylist=new Solution();
  for(i=0;i<T;i++){
      var data=parseInt(readLine());
      head=mylist.insert(head,data);
  }
  head=mylist.removeDuplicates(head);
  mylist.display(head);
}
